File: backend/app/crud/task_crud.py
from sqlalchemy.orm import Session
from app import models, schemas

def create_task(db: Session, task: schemas.TaskCreate):
    db_task = models.Task(title=task.title)
    db.add(db_task)
    db.commit()
    db.refresh(db_task)
    return db_task

def get_tasks(db: Session):
    return db.query(models.Task).all()

def delete_task(db: Session, task_id: int):
    task = db.get(models.Task, task_id)
    if task:
        db.delete(task)
        db.commit()

def complete_task(db: Session, task_id: int):
    task = db.get(models.Task, task_id)
    if task:
        task.completed = True
        db.commit()
        db.refresh(task)
    return task

File: backend-node/src/crud/taskCrud.ts
import { AppDataSource } from "../database/data-source";
import { Task } from "../entities/Tasks";
import { CreateTaskDto, UpdateTaskDto } from "../schemas/taskSchemas";

const taskRepository = AppDataSource.getRepository(Task);

// CREATE: Crear una tarea
export const createTask = async (taskData: CreateTaskDto): Promise<Task> => {
    const dbTask = taskRepository.create({
        title: taskData.title,
        completed: false
    });

    await taskRepository.save(dbTask);
    return dbTask;
};

// READ ALL: Obtener todas las tareas
export const getTasks = async (): Promise<Task[]> => {
    return await taskRepository.find({
        order: { id: "ASC" }
    });
};

// READ ONE: Obtener una tarea por ID
export const getTaskById = async (taskId: number): Promise<Task | null> => {
    return await taskRepository.findOneBy({ id: taskId });
};